src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -903,11 +905,14 @@ private ISet<string> CalculateFilesToExcludeInPack(IPackTaskRequest<IMSBuildItem
 
         private static void InitializePackageDependencies(
             LockFile assetsFile,
+            PackageBuilder packageBuilder,
             Dictionary<NuGetFramework, HashSet<LibraryDependency>> dependenciesByFramework,
             ISet<NuGetFramework> frameworkWithSuppressedDependencies)
         {
+            StringBuilder nowarnProperties = new();
```

Comment:
Have you checked how restore handles per-package nowarn properties?  I have no idea, but using a string builder to make a giant string, and then parsing it again seems tremendously inefficient. I imagine this isn't a particularly hot path, so low performance nowarn passing isn't the end of the world, I'm just curious if restore does the same, or if it takes advantage of better data structures.